node/src/exchange/client.rs

```diff
@@ -58,13 +76,79 @@ impl ExchangeClientHandler {
 
         let req_id = req_resp.send_request(&peer, request);
         self.reqs.insert(req_id, ReqInfo { amount, respond_to });
+    }
 
-        trace!("Request initiated");
+    fn send_head_request(
```

Comment:
is there a way to test that?

Reply:
not, currently. we need to implement peer finding and connect to a network with multiple peers.

Reply:
actually i thought a way, i will implement it tomorrow

Reply:
tests are added